Backend test flake: mis-ordering of exploration states

Example: https://github.com/oppia/oppia/actions/runs/3476372502/jobs/5811542387#step:8:1949

Log:

```
======================================================================
FAIL: test_verify_all_states_reachable (core.domain.exp_domain_test.ExplorationDomainUnitTests)
----------------------------------------------------------------------
core.utils.ValidationError: Please fix the following issues before saving this exploration: 1. The following states are not reachable from the initial state: End 2. It is impossible to complete the exploration from the following states: Introduction, Stuck State 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/oppia/oppia/core/domain/exp_domain_test.py", line 4206, in test_verify_all_states_reachable
    exploration.validate(strict=True)
AssertionError: "Please fix the following issues before saving this exploration: 1. The following states are not reachable from the initial state: End 2. It is impossible to complete the exploration from the following states: Stuck State, Introduction" does not match "Please fix the following issues before saving this exploration: 1. The following states are not reachable from the initial state: End 2. It is impossible to complete the exploration from the following states: Introduction, Stuck State "
```

We likely need to change our assertion so that the order of states does not matter.

--- a/core/domain/exp_domain.py
+++ b/core/domain/exp_domain.py
@@ -2098,8 +2098,8 @@ class Exploration(translation_domain.BaseTranslatableObject):
                             curr_queue.append(dest_if_stuck_state)
 
         if len(self.states) != len(processed_queue):
-            unseen_states = list(
-                set(self.states.keys()) - set(processed_queue))
+            unseen_states = sorted(list(
+                set(self.states.keys()) - set(processed_queue)))
             raise utils.ValidationError(
                 'The following states are not reachable from the initial '
                 'state: %s' % ', '.join(unseen_states))

